gis/grass/Portfile

```diff
@@ -1,11 +1,300 @@
+    }
+
+    post-destroot {
+        copy ${worksrcpath}/${app_name} ${destroot}${applications_dir}
+    }
+}
+
+variant mysql5 description {Add MySQL 5 support} {
+    depends_lib-append      path:bin/mysql_config5:mysql5
+    configure.args-append   --with-mysql
+    configure.args-append   --with-mysql-includes=${prefix}/include/mysql5/mysql
+    configure.args-append   --with-mysql-libs=${prefix}/lib/mysql5/mysql
+    configure.env-append    MYSQLD_CONFIG=${prefix}/lib/mysql5/bin/mysql_config
+}
+
+# PostGreSQL variants (from the GDAL port)
+set postgresql_suffixes {10 11 12 13}
+
+set postgresql_variants {}
+foreach suffix ${postgresql_suffixes} {
+    lappend postgresql_variants postgresql${suffix}
+}
+
+foreach suffix ${postgresql_suffixes} {
+    set vrt postgresql${suffix}
+    set pgversion [string index ${suffix} 0].[string index ${suffix} 1]
+    set index [lsearch -exact ${postgresql_variants} ${vrt}]
+    set conf [lreplace ${postgresql_variants} ${index} ${index}]
+
+    variant ${vrt} description "Use PostgreSQL ${pgversion}" conflicts {*}${conf} "
+        depends_lib-append      port:${vrt}
+        configure.args-append   --with-postgres-includes=${prefix}/include/${vrt}
+        configure.args-append   --with-postgres-libs=${prefix}/lib/${vrt}
+        configure.args-delete   --with-postgres=no
+        configure.args-append   --with-postgres=yes
+    "
+}
+
+# PostGreSQL default
+set pgdefault "if {"
+
+foreach suffix ${postgresql_suffixes} {
+    set pgdefault "${pgdefault}!\[variant_isset postgresql${suffix}\] && "
+}
+
+set pgdefault [string range ${pgdefault} 0 end-4]
+set pgdefault "${pgdefault}} { default_variants +postgresql${suffix} }"
+
+eval ${pgdefault}
+
+variant sqlite description {Add SQLite 3 support} {
+    depends_lib-append    port:sqlite3
+    configure.args-append --with-sqlite
+    configure.args-append --with-sqlite-includes=${prefix}/include
+    configure.args-append --with-sqlite-libs=${prefix}/lib
+}
+
+variant openblas description {Use OpenBLAS for BLAS/LAPACK} {
+    depends_lib-append      port:openblas
+
+    configure.args-delete   --with-blas=no
+    configure.args-delete   --with-lapack=no
+    configure.args-append   --with-blas=yes
+    configure.args-append   --with-lapack=yes
+    configure.args-append   --with-blas-includes=${prefix}/include
+    configure.args-append   --with-libblas=${prefix}/lib/libopenblas.dylib
+    configure.args-append   --with-lapack-includes=${prefix}/include
+    configure.args-append   --with-liblapack=${prefix}/lib/libopenblas.dylib
+}
+
+post-configure {
+    reinplace -E "s|\\\$\\\(CPPFLAGS)| \$(CPPFLAGS) \
+        \\\\\"-Drestrict=\\\\\" \\\\\"-D__attribute__(x)=\\\\\" \\\\\"-D_Nonnull=\\\\\" \
+        \\\\\"-Dint8_t=char\\\\\" \\\\\"-DCF_INLINE=\\\\\" \\\\\"-D_Null_unspecified=\\\\\" \
+        \\\\\"-D__DARWIN_OS_INLINE=\\\\\"|" ${worksrcpath}/python/libgrass_interface_generator/Makefile
```

Comment:
This reinplace is difficult to comprehend and there's no indication here of why it's happening. If you're fixing some bug, add a comment with the URL to the bug report.

Since you're not using any MacPorts variables in the reinplace, you could save yourself a lot of escaping by using `{}` string delimiters instead of `""`.

Reply:
It looks like what this reinplace is trying to is, for each occurrence of `$(CPPFLAGS)`, add some stuff after that. The better way to do that would be to find where `CPPFLAGS` is set and add those things there. That would be only one change needed, vs. however many occurrences of `$(CPPFLAGS)` there are. And then instead of a reinplace it could be implemented as a patchfile which is highly preferable.

Reply:
These are for passing to the Python package ctypesgen, which generates python interface of C code. Ctypesgen in turn send these to compiler preprocessing (eg. `clang -E ...`), which result is used for further parsing. The ctypesgen parser does not recognise all C code yet (such as keywords like `restrict` or `__attribute__(x)` statements) and throws several thousands lines of error messages. These are now mainly (probably only) noise, but I'm working upstreams to at least reduce them to a minimum. Meanwhile, I suggest to leave it as it has been in grass7.